src/main/java/org/openstreetmap/atlas/geography/Polygon.java

```diff
@@ -255,17 +255,15 @@ public boolean fullyGeometricallyEncloses(final Rectangle rectangle)
             return false;
         }
         // The item is within the bounds of this Polygon
-        // if this value overflows, use JTS to correctly calculate covers
-        if (awtOverflows())
+        // if this value overflows, or if the much faster AWT returns false,
+        // use JTS to correctly calculate covers
+        if (awtOverflows() || !awtArea().contains(rectangle.asAwtRectangle()))
```

Comment:
Here that would fall back to JTS for all cases where the area does not contain the rectangle, even when the area is half a world away from the rectangle...
I would fall back only if the area's bounds overlap with the rectangle, and the awt contains is false.

Reply:
wait maybe I'm confused, doesn't that happen on line 252 already?